lib/charms/openfga_k8s/v1/openfga.py

```diff
@@ -0,0 +1,423 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""# Interface Library for OpenFGA.
+
+This library wraps relation endpoints using the `openfga` interface
+and provides a Python API for requesting OpenFGA authorization model
+stores to be created.
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.openfga_k8s.v1.openfga
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+  openfga:
+    interface: openfga
```

Comment:
i find it interesting that we don't include the major version of the interface in the interface name.. what guarantees does the requiring charm have that a providing charm using the right major version of the library will be related/integrated with it?

Reply:
That's a good point. It has no guarantee, but tbh I think that this should be handled by juju itself.

Reply:
But there are no guarantees that nothing will break between minor versions either (it has happened)....

Reply:
well.. not by juju.. juju just deploys and runs the charm.. it doesn't care which framework was used to write the charm and so it doesn't care about libraries and charm relation interfaces, either..

Reply:
and also - any breaking change should result in a new major version.. minor versions are expected to maintain compatibility

Reply:
> and also - any breaking change should result in a new major version.. minor versions are expected to maintain compatibility

I agree with that, I am just saying that you don't get any guarantees for it, other than the maintainer's word. I guess your point is that even if the maintainer keeps their word and bump the major version in a breaking change, it's still not very useful from a charm's perspective because if your libs are not up to date everything will crash. But I still think that this is a design problem with charm libs and shouldn't be handled by the charms themselves (if that makes sense).

> juju just deploys and runs the charm

Agreed, but AFAICT libs are first class citizens in the charm world (else why would there be a `lib` section for each charm on charmhub). So I think that this should be handled on another level (ops framework or juju)